File: backend/warranty/src/main/java/edu/uth/warranty/model/Staff.java
package edu.uth.warranty.model;

import jakarta.persistence.*;
import edu.uth.warranty.common.Role;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "Staff")
public class Staff {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long staffId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "center_id", nullable = false)
    private ServiceCenter center;

    @Column(name = "name", nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(name = "role", nullable = false)
    private Role role;

    @Column(name = "phone", nullable = false, unique = true)
    private String phone;

    @Column(name = "email", nullable = false, unique = true)
    private String email;

    @Column(name = "username", nullable = false)
    private String username;

    @Column(name = "password", nullable = false)
    private String password;
    
}

File: frontend/src/types/staff.ts
import { UserRole } from "./auth";

export interface StaffRequest {
    id?: number;
    centerId: number;
    name: string;
    role: UserRole;
    phone: string;
    address?: string; 
    email: string;
    username: string;
    password?: string; 
}

export interface StaffResponse {
    id: number;
    centerId: number;
    centerName: string;
    name: string;
    role: UserRole;
    phone: string;
    address: string;
    email: string;
    username: string;
}